### PR TITLE
SOLR-17494: Remove language specific writer types

### PR DESCRIPTION
**Pull request by epugh**
https://issues.apache.org/jira/browse/SOLR-17494

# Description

Language specific writers types are "evaled" which isn't secure.  And not a pattern to be used.

# Solution

Remove them from main.

### CODE REVIEW

File: solr/core/src/test/org/apache/solr/response/JSONWriterTest.java [hunk 1]
```diff
@@ -53,24 +53,17 @@ private void jsonEq(String expected, String received) {
...
 
   @Test
-  public void testTypes() throws IOException {
+  public void testSimpleJson() throws IOException {
     SolrQueryRequest req = req("q", "dummy", "indent", "off");
     SolrQueryResponse rsp = new SolrQueryResponse();
-    QueryResponseWriter w = new PythonResponseWriter();
+    QueryResponseWriter w = new JSONResponseWriter();
+    new StringWriter();
+    StringWriter buf;
```
Comment (gerlowskija) [inline solr/core/src/test/org/apache/solr/response/JSONWriterTest.java:61]: [-1] I think L60-L61 is a typo and should be the single line:

```
StringWriter buf = new StringWriter();
```

**Review by gerlowskija — COMMENTED** [review]
One test-side typo that needs fixed, but when that's fixed this LGTM.

Oh, and maybe this should have a 10.0 CHANGES.txt entry?  (Or maybe two entries - one for 9.8 to signal the deprecation, and one for 10.0 to announce the removal?)

**Comment by epugh** [comment]
Thoughts on how to mark it "deprecated" in 9x?   In some places it makes sense to put out a log message.   What would being deprecated mean here?   Just a PR that adds `@deprecated` in the `branch_9x` branch?

**Comment by janhoy** [comment]
Deprecate as notice in ref guide. Also we have a separate deprecation logging mechanism.

**Comment by epugh** [comment]
Can I get a LGTM and then I'll add CHANGES.txt and merge....

**Comment by gerlowskija** [comment]
Woohoo, thanks @epugh !

**Pull request merged**